spec.md

```diff
@@ -99,7 +97,8 @@ The `Span` interface must have the following capabilities:
 
 The `Tracer` interface must have the following capabilities:
 
-- Start a `Span` that has no parent, commonly referred to as a *root* `Span` **(py: `start_trace`, go: `StartTrace`)**
+- Start a new child `Span` with a given operation name. With no parent, commonly referred to as a *root* `Span`, no trace attributes exist. With a parent any trace attributes must be passed through into the created child. **(py: `create_span`, go: `CreateSpan`)**
```

Comment:
@dkuebric If i've understood you correctly then these things can be done already with the existing API without needing more than the create_span method.

`create_span` covers the later two scenarios (create_trace is not needed). 

`join_trace` is just about obtaining the span from the context. You don't need to join the trace, when you've called `span = tracer.getSpanExtractor(request.class).extract(request)` then this span is already part of the trace and you're free to start using it.

The benefit of this approach is that it removes the notion of a Trace object completely from the API and the schema the user needs to load into their mind to understand how to use that API. That is the trace is just something invisible in the context that the user is never concerned about.